api/internal/handler/upstream/upstream_test.go

```diff
@@ -19,285 +19,1719 @@ package upstream
+						},
+					},
+				},
+			},
+			wantRet: &entity.Upstream{
+				BaseInfo: entity.BaseInfo{
+					ID: "u1",
+				},
+				UpstreamDef: entity.UpstreamDef{
+					Name: "upstream1",
+					Timeout: map[string]interface{}{
+						"connect": 15,
+						"send":    15,
+						"read":    15,
+					},
+					Checks: map[string]interface{}{
+						"active": map[string]interface{}{
+							"timeout":   float64(5),
+							"http_path": "/status",
+							"host":      "foo.com",
+							"healthy": map[string]interface{}{
+								"interval":  2,
+								"successes": 1,
+							},
+							"unhealthy": map[string]interface{}{
+								"interval":      1,
+								"http_failures": 2,
+							},
+							"req_headers": []interface{}{"User-Agent: curl/7.29.0"},
+						},
+						"passive": map[string]interface{}{
+							"healthy": map[string]interface{}{
+								"http_statuses": []interface{}{float64(200), float64(201)},
+								"successes":     float64(3),
+							},
+							"unhealthy": map[string]interface{}{
+								"http_statuses": []interface{}{float64(500)},
+								"http_failures": 3,
+								"tcp_failures":  3,
+							},
+						},
+					},
+					Key: "server_addr",
+					Nodes: []map[string]interface{}{
+						{
+							"host":   "39.97.63.215",
+							"port":   float64(80),
+							"weight": float64(1),
+						},
+					},
+				},
+			},
+		},
+		{
+			caseDesc: "create failed, different id",
+			giveInput: &UpdateInput{
+				ID: "u1",
+				Upstream: entity.Upstream{
+					BaseInfo: entity.BaseInfo{
+						ID: "u2",
+					},
+					UpstreamDef: entity.UpstreamDef{
+						Name: "upstream1",
+						Timeout: map[string]interface{}{
+							"connect": 15,
+							"send":    15,
+							"read":    15,
+						},
+						Checks: map[string]interface{}{
+							"active": map[string]interface{}{
+								"timeout":   float64(5),
+								"http_path": "/status",
+								"host":      "foo.com",
+								"healthy": map[string]interface{}{
+									"interval":  2,
+									"successes": 1,
+								},
+								"unhealthy": map[string]interface{}{
+									"interval":      1,
+									"http_failures": 2,
+								},
+								"req_headers": []interface{}{"User-Agent: curl/7.29.0"},
+							},
+							"passive": map[string]interface{}{
+								"healthy": map[string]interface{}{
+									"http_statuses": []interface{}{float64(200), float64(201)},
+									"successes":     float64(3),
+								},
+								"unhealthy": map[string]interface{}{
+									"http_statuses": []interface{}{float64(500)},
+									"http_failures": 3,
+									"tcp_failures":  3,
+								},
+							},
+						},
+						Key: "server_addr",
+						Nodes: []map[string]interface{}{
+							{
+								"host":   "39.97.63.215",
+								"port":   float64(80),
+								"weight": float64(1),
+							},
+						},
+					},
+				},
+			},
+			wantRet: &data.SpecCodeResponse{StatusCode: http.StatusBadRequest},
+			wantErr: fmt.Errorf("ID on path (u1) doesn't match ID on body (u2)"),
+		},
```

Comment:
Maybe we need to test `Update` failed.

Reply:
Update I added a test case that could not be updated due to ID matching failure, but my description is wrong and has been fixed.